### PR TITLE
Propagate usage of controller-runtime client

### PR DESCRIPTION
**Pull request by ialidzhikov**
**What this PR does / why we need it**:
- Propagate usage of controller-runtime client
- Update logging stack deletion to use label selector

/king cleanup

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: pkg/operation/common/utils.go [hunk 1]
```diff
@@ -427,61 +434,52 @@ func InjectCSIFeatureGates(kubeVersion string, featureGates map[string]bool) (ma
...
-		return err
-	}
-	if err := k8sClient.DeleteServiceAccount(namespace, "fluent-bit"); err != nil && !apierrors.IsNotFound(err) {
+	requirement, err := labels.NewRequirement(GardenRole, selection.Equals, []string{GardenRoleLogging})
+	if err != nil {
 		return err
 	}
-	if err := k8sClient.DeleteHorizontalPodAutoscaler(namespace, "fluentd-es"); err != nil && !apierrors.IsNotFound(err) {
-		return err
+	listOptions := &client.ListOptions{
```
Comment (adracus) [inline pkg/operation/common/utils.go]: You can instantiate the `ListOptions` more 'fluently' by using `client.InNamespace(namespace).LabelSelector ...`

File: pkg/operation/common/utils.go [hunk 2]
```diff
@@ -427,61 +434,52 @@ func InjectCSIFeatureGates(kubeVersion string, featureGates map[string]bool) (ma
...
+		Namespace:     namespace,
+		LabelSelector: labels.NewSelector().Add(*requirement),
 	}
-	for _, name := range services {
-		if err := k8sClient.DeleteService(namespace, name); err != nil && !apierrors.IsNotFound(err) {
+
+	for _, list := range lists {
+		if err := k8sClient.Client().List(context.TODO(), listOptions, list); err != nil {
```
Comment (adracus) [inline pkg/operation/common/utils.go:463]: Please add a TODO here that we should switch to `DeleteCollection` once it's finally there kubernetes-sigs/controller-runtime#324

**Review by adracus — COMMENTED** [review]
Minor nits, otherwise very nice!

File: pkg/operation/seed/seed.go [hunk 3]
```diff
@@ -315,7 +320,8 @@ func BootstrapCluster(seed *Seed, secrets map[string]*corev1.Secret, imageVector
 	)
 
 	if vpaEnabled {
-		existingSecrets, err := k8sSeedClient.ListSecrets(common.GardenNamespace, metav1.ListOptions{})
+		existingSecrets := &corev1.SecretList{}
+		err = k8sSeedClient.Client().List(context.TODO(), client.InNamespace(common.GardenNamespace), existingSecrets)
```
Comment (timuthy) [inline pkg/operation/seed/seed.go]:
```suggestion
		if err = k8sSeedClient.Client().List(context.TODO(), client.InNamespace(common.GardenNamespace), existingSecrets); err != nil ...
```

File: pkg/operation/seed/seed.go [hunk 4]
```diff
@@ -298,12 +301,14 @@ func BootstrapCluster(seed *Seed, secrets map[string]*corev1.Secret, imageVector
...
 
-		clusterIssuer, err = createClusterIssuer(k8sSeedClient, certificateManagement)
+		clusterIssuer, err = createClusterIssuer(certificateManagement)
 		if err != nil {
 			return fmt.Errorf("cannot create Cluster Issuer for certificate management: %v", err)
 		}
 	} else {
-		if err := k8sSeedClient.DeleteDeployment(common.GardenNamespace, common.CertManagerResourceName); err != nil && !apierrors.IsNotFound(err) {
+		obj := &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Name: common.CertManagerResourceName, Namespace: common.GardenNamespace}}
+		err = k8sSeedClient.Client().Delete(context.TODO(), obj, kubernetes.DefaultDeleteOptionFuncs...)
```
Comment (timuthy) [inline pkg/operation/seed/seed.go]:
```suggestion
		if err = k8sSeedClient.Client().Delete(context.TODO(), obj, kubernetes.DefaultDeleteOptionFuncs...); err != nil && !apierrors.IsNotFound(err) ...
```

File: pkg/operation/seed/seed.go [hunk 5]
```diff
@@ -249,16 +251,17 @@ func BootstrapCluster(seed *Seed, secrets map[string]*corev1.Secret, imageVector
...
 		kibanaHost          string
 		fluentdReplicaCount int32
 		loggingEnabled      = controllermanagerfeatures.FeatureGate.Enabled(features.Logging)
+		existingSecretsMap  = map[string]*corev1.Secret{}
 	)
 
 	if loggingEnabled {
-		existingSecrets, err := k8sSeedClient.ListSecrets(common.GardenNamespace, metav1.ListOptions{})
+		existingSecrets := &corev1.SecretList{}
+		err = k8sSeedClient.Client().List(context.TODO(), client.InNamespace(common.GardenNamespace), existingSecrets)
```
Comment (timuthy) [inline pkg/operation/seed/seed.go]:
```suggestion
		if err = k8sSeedClient.Client().List(context.TODO(), client.InNamespace(common.GardenNamespace), existingSecrets); err != nil ...
```

File: pkg/operation/seed/seed.go [hunk 6]
```diff
@@ -64,7 +65,8 @@ var wantedCertificateAuthorities = map[string]*utilsecrets.CertificateSecretConf
 // New takes a <k8sGardenClient>, the <k8sGardenInformers> and a <seed> manifest, and creates a new Seed representation.
 // It will add the CloudProfile and identify the cloud provider.
 func New(k8sGardenClient kubernetes.Interface, k8sGardenInformers gardeninformers.Interface, seed *gardenv1beta1.Seed) (*Seed, error) {
-	secret, err := k8sGardenClient.GetSecret(seed.Spec.SecretRef.Namespace, seed.Spec.SecretRef.Name)
+	secret := &corev1.Secret{}
+	err := k8sGardenClient.Client().Get(context.TODO(), kutil.Key(seed.Spec.SecretRef.Namespace, seed.Spec.SecretRef.Name), secret)
```
Comment (timuthy) [inline pkg/operation/seed/seed.go]:
```suggestion
	if err := k8sGardenClient.Client().Get(context.TODO(), kutil.Key(seed.Spec.SecretRef.Namespace, seed.Spec.SecretRef.Name), secret); err != nil ...
```

File: pkg/operation/seed/seed.go [hunk 7]
```diff
@@ -337,22 +343,23 @@ func BootstrapCluster(seed *Seed, secrets map[string]*corev1.Secret, imageVector
...
-	}
-	if err := k8sSeedClient.DeleteSecret(common.GardenNamespace, "gardener-external-admission-controller-tls"); err != nil && !apierrors.IsNotFound(err) {
-		return err
+	objects := []runtime.Object{
+		&corev1.Service{ObjectMeta: metav1.ObjectMeta{Name: "gardener-external-admission-controller", Namespace: common.GardenNamespace}},
+		&appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Name: "gardener-external-admission-controller", Namespace: common.GardenNamespace}},
+		&corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "gardener-external-admission-controller-tls", Namespace: common.GardenNamespace}},
+	}
+	for _, object := range objects {
+		err := k8sSeedClient.Client().Delete(context.TODO(), object, kubernetes.DefaultDeleteOptionFuncs...)
```
Comment (timuthy) [inline pkg/operation/seed/seed.go]:
```suggestion
		if err := k8sSeedClient.Client().Delete(context.TODO(), object, kubernetes.DefaultDeleteOptionFuncs...); err != nil && !apierrors.IsNotFound(err) ...
```

**Comment by ialidzhikov** [comment]
Thank you! Applied the suggestions.

**Review by timuthy — APPROVED** [review]
/lgtm

**Pull request merged**